### DISCUSSION TITLE
How many neutrino nodes can be connected to a single bitcoin core

### DISCUSSION THREAD

**Discussion by youngzhenhao** [lightningnetwork/lnd#9280] [Q&A] [answered]
I configed `lnd.neutrino.connect=xxx.xxx.xxx.xxx:8333` in lit.conf, and the connected server's ports 8333 and 8334 are accessible, but sometimes the logs appear a lot of `[INF] BTCN: Lost peer xxx.xxx.xxx.xxx:8333 (outbound)`. I am wondering if it is because there are too many neutrino nodes using that server and what the connection limit is and is it possible to config in bitcoin.conf.

**Comment by guggero** [comment, marked as answer]
You define that with `maxconnections=<n>` in your `bitcoind.conf` file. The default seems to be 125, so quite a lot.
But be aware that if your `bitcoind`'s port 8333 is open to the internet, it will allow other nodes (and Neutrino clients) to use that, sharing the 125 connections. If you don't want that, I think you'd need to set up a firewall rule to only allow your own Neutrino client. Or maybe there are other ways to restrict access to the 8333 port.

**Reply by youngzhenhao** [reply]
If there are 10000 neutrino nodes, is it possible to have them connect to the same bitcoin core, or would it be better to have them each connect to 5 different bitcoin core nodes, i.e. 2000 neutrino nodes per bitcoin core.

**Reply by guggero** [reply]
Uff, these are pretty large numbers and you might get more useful answers over at the https://github.com/bitcoin/bitcoin repository...
My guess is it probably mostly comes down to how much resources you have available to give to `bitcoind`. Serving compact block filters to that many client will likely put quite some load onto that node, probably most notably on the disk.
So as long as you have super fast SSDs available I would _guess_ it should be possible.
But if you have the possibility, my gut feeling is that you probably run into fewer issues if you balance the load across multiple `bitcoind` nodes. But I don't have any concrete numbers to base that on.
